base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
```

Comment:
- [ ] Correct name method is not properly overridden. `correct_name` here should return the result of the `correct_name` of `@namable`. This should be similar to:

```ruby
class Decorator < Nameable
    attr_accessor :nameable
  
  ...

    def correct_name
      @nameable.correct_name
    end
end
```